archlinux/Makefile

```diff
@@ -0,0 +1,102 @@
+#rootfs for Archlinux ARM
+
+SRCDIR=src
+BUILDDIR=build
+CUSTOMIZATION=customization
+
+SUDO=/usr/bin/sudo
+QEMU=/usr/bin/qemu-arm-static
+QEMU64=/usr/bin/qemu-aarch64-static
+
+GNULINUX_OTA_ARCH=armv7
+GNULINUX_SYSTEM_IMAGE_FILE=ArchLinuxARM-$(GNULINUX_OTA_ARCH)-latest.tar.gz
+GNULINUX_SYSTEM_IMAGE_URL=https://archlinuxarm.org/os/$(GNULINUX_SYSTEM_IMAGE_FILE)
+
+SRC_GNULINUX_SYSTEM_IMAGE_FILE=$(SRCDIR)/$(GNULINUX_SYSTEM_IMAGE_FILE)
+
+GNULINUX_ROOTFS=halium.rootfs.tar.gz
+
+all: $(GNULINUX_ROOTFS)
+
+$(GNULINUX_ROOTFS): $(SUDO) $(BUILDDIR) .rootfs
+	$(info Building $(GNULINUX_ROOTFS))
+	@$(SUDO) tar czf $@ -C $(BUILDDIR) .
+	@echo "Completed: $(GNULINUX_ROOTFS)"
+
+$(SRC_GNULINUX_SYSTEM_IMAGE_FILE): $(SRCDIR)
+	$(info Downloading GNU/Linux Image: $(GNULINUX_SYSTEM_IMAGE_FILE))
+	@curl -L $(GNULINUX_SYSTEM_IMAGE_URL) -o $@
+
+.extract: $(SUDO) $(BUILDDIR) $(SRC_GNULINUX_SYSTEM_IMAGE_FILE)
+	$(info Extracting the archive $(GNULINUX_SYSTEM_IMAGE_FILE))
+	@$(SUDO) tar --numeric-owner -xzf $(SRC_GNULINUX_SYSTEM_IMAGE_FILE) -C $(BUILDDIR) 2> /dev/null
+	@touch .extract
+
+.mount: .extract mount
+	@touch .mount
+
+.umount: .mount umount
+	@touch .umount
+
+.patch-rootfs: $(SUDO) $(BUILDDIR) .mount
+	$(info Patching rootfs inside the chroot)
+	@$(SUDO) chroot $(BUILDDIR) /bin/sh /root/$(CUSTOMIZATION)/chroot-patch-rootfs.sh
+	@touch .patch-rootfs
+
+.rootfs: .patch-rootfs .umount
+	@touch .rootfs
+
+.mount-manual: $(SUDO) $(QEMU) $(QEMU64) $(BUILDDIR) $(CUSTOMIZATION) .extract
```

Comment:
Because the current build of archlinuxARM rootfs can be done on any other distribution (I have avoid to use archlinux packages on the host)
The same Makefile can be used if needed for some other distribution (eg: Gentoo)